### PR TITLE
Enabled memproof on devnet

### PR DESCRIPTION
**Pull request by zeeshan595**
Enabled memproof on dev net by following joel's instructions.
https://www51.v1host.com/Holo/story.mvc/Summary?oidToken=Story:43508

### CODE REVIEW

**Comment by alastairong** [comment]
I don't understand this PR. Why are you changing the public key expected from the alpha-membrane-proof service?

@zo-el

**Review by zo-el — CHANGES_REQUESTED** [review]
@zeeshan595 where is the work to pass the right agent keys to mem-proof, for `devNet` that we spoke about?

**Comment by zeeshan595** [comment]
> @zeeshan595 where is the work to pass the right agent keys to mem-proof, for `devNet` that we spoke about?

Added the work here https://github.com/Holo-Host/hpos-configure-holochain/pull/129
I need to update the reference (sha) from this project but I am waiting on review for the other work in case I have to modify it :)

CC @alastairong

**Comment by peeech** [comment]
Now hash of a commit, sha and cargoSha need to be updated in overlays. Lemme know @zeeshan595 if you need any help with that

**Review by alastairong — APPROVED** [review]

**Comment by zeeshan595** [comment]
Not sure why nightly tests are failing it was saying it timed when testing `holochain-api`

```
2 failing
  1) Test of functionality of holo network from UI to Holoport
       "before all" hook for "Sign up and connect to host":
     Error: Master holoport 172.26.72.248 failed to reset - cannot continue
      at publishHappOn0 (tests/dummy-ui-utils.js:66:37)
      at exports.setUpHoloports (tests/dummy-ui-utils.js:132:15)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
      at async Context.<anonymous> (tests/dummy-ui-test.js:33:13)
  2) Test of host console UI functions
       Check if calls to holochain-api endpoint are successful:
     TimeoutError: Timeout exceeded while waiting for event
      at Timeout.<anonymous> (node_modules/puppeteer/lib/cjs/puppeteer/common/helper.js:120:28)
      at listOnTimeout (internal/timers.js:554:17)
      at processTimers (internal/timers.js:497:7)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 2.
```

**Comment by alastairong** [comment]
@zeeshan595 failing because:

```
Jun 08 16:53:23 hpos configure-holochain[100569]: Jun 08 16:53:23.424 ERROR run:install_and_activate_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}:register_and_install_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}: configure_holochain::websocket: error=error: InternalError("Conductor returned an error while using a ConductorApi: GenesisFailed { errors: [ConductorApiError(WorkflowError(GenesisFailure(\"Joining code invalid: incorrect registered agent key\")))] }")
```

Which means that configure-holochain is still trying to use the real devnet membrane proof rather than requesting a replacement for its real agentKey

**Comment by zo-el** [comment]
So I had to remove the old mem-proof and agent-key file from the holoports and then reran the tests for the nightly tests to pass.

My suggestion is that someone runs the nightly tests on a different holoport and see if the tests run just by restarting the holoport or do we have to delete the pub-key file.

**Comment by peeech** [comment]
I have run nightly test on a holoport that was running develop previously. For this test I set PR 1266.

Test failed during holoport configuration:
```
× configure-holochain.service
     Loaded: loaded (/etc/systemd/system/configure-holochain.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Mon 2022-06-13 11:40:50 UTC; 4s ago
    Process: 317497 ExecStartPre=/nix/store/hgcfig880k7b8q8chl5m3wipd5cjq789-unit-script-configure-holochain-pre-start/bin/configure-holochain-pre-start (code=exited, status=0/SUCCESS)
    Process: 317504 ExecStart=/nix/store/82qfiiavjhqyygzsjxfnnjwh34mdhq34-hpos-configure-holochain/bin/configure-holochain /var/lib/configure-holochain/config.yaml (code=exited, status=1/FAILURE)
   Main PID: 317504 (code=exited, status=1/FAILURE)
         IP: 2.2M in, 55.6K out
        CPU: 113ms

Jun 13 11:40:47 hpos configure-holochain[317504]: Jun 13 11:40:47.167 DEBUG run:install_and_activate_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}:register_and_install_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}:download_file: configure_holochain: downloading
Jun 13 11:40:47 hpos configure-holochain[317504]: Jun 13 11:40:47.720 DEBUG run:install_and_activate_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}:register_and_install_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}:download_file: configure_holochain: download successful
Jun 13 11:40:47 hpos configure-holochain[317504]: Jun 13 11:40:47.721  INFO run:install_and_activate_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}:register_and_install_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}: configure_holochain::websocket: using uid to install: 1e0b99a2-f828-41a5-b410-33e81ee6f126
Jun 13 11:40:50 hpos configure-holochain[317504]: Jun 13 11:40:50.198 ERROR run:install_and_activate_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}:register_and_install_happ{properties=Some(YamlProperties(Mapping(Mapping { map: {String("skip_proof"): Bool(false), String("holo_agent_override"): String("uhCAkwovtK1wVMgH_SNKbiD7Mu1kyK2O7decaAZ9FPFeHbTKP1pXd")} })))}: configure_holochain::websocket: error=error: InternalError("Conductor returned an error while using a ConductorApi: GenesisFailed { errors: [ConductorApiError(WorkflowError(GenesisFailure(\"Joining code invalid: incorrect registered agent key\")))] }")
Jun 13 11:40:50 hpos configure-holochain[317504]: Jun 13 11:40:50.198 ERROR run: configure_holochain: error=error: InternalError("Conductor returned an error while using a ConductorApi: GenesisFailed { errors: [ConductorApiError(WorkflowError(GenesisFailure(\"Joining code invalid: incorrect registered agent key\")))] }")
Jun 13 11:40:50 hpos configure-holochain[317504]: Error: error: InternalError("Conductor returned an error while using a ConductorApi: GenesisFailed { errors: [ConductorApiError(WorkflowError(GenesisFailure(\"Joining code invalid: incorrect registered agent key\")))] }")
Jun 13 11:40:50 hpos systemd[1]: configure-holochain.service: Main process exited, code=exited, status=1/FAILURE
Jun 13 11:40:50 hpos systemd[1]: configure-holochain.service: Failed with result 'exit-code'.
Jun 13 11:40:50 hpos systemd[1]: Failed to start configure-holochain.service.
```

**Comment by alastairong** [comment]
I have a feeling you and Joel should pair on this to get it over the time. Perhaps there's something about the design of this approach that was missed?

**Comment by alastairong** [comment]
@zo-el ping for review. Once reviewed I'll run nightly test properly

File: README.md [hunk 1]
```diff
@@ -180,3 +179,8 @@ To run nightly tests for you PR do the following:
 
 #### NOTE: Nightly tests should always use workflow from `develop` branch
 
+## Enabling Membrane Proof on DevNet
+
+Membrane proof is requested on `devNet` when a new agent pub key is [created](https://github.com/Holo-Host/hpos-configure-holochain/blob/develop/src/websocket.rs#L108). 
+
+If an already existing agent pub key is used the above mentioned step will be skipped. This means, the membrane proof might not be correct. To resolve this you can either remove your agent pub key to generate a new one or update the UUID causing a new agent pub key to be generated.
```
Comment (alastairong) [inline README.md:186]: Bumping the UUID doesn't create a new agent pub key but I think it triggers a new membrane proof download anyway?

Comment (zo-el) [inline README.md:186]:
```suggestion
If an already existing agent pub key is used the above mentioned step will be skipped. This means the membrane proof might not be correct. To resolve this you can either remove your agent pub key to generate a new one or update the UID causing a new agent pub key to be generated.
```

File: overlays/holo-nixpkgs/configure-holochain/default.nix [hunk 2]
```diff
@@ -7,11 +7,11 @@ rustPlatform.buildRustPackage {
   src = fetchFromGitHub {
     owner = "Holo-Host";
     repo = "hpos-configure-holochain";
-    rev = "1416715372814dc7a8ef953239f2b2d2e326ac51"; # Bumped to use holochain v0.0.131
-    sha256 = "sha256-XKrUzkw7IVtlbhOR//yudz5iEsVRjcx/qkSg2CcmlAQ=";
+    rev = "aa64c168f8c13a6618bc6362e3a2688f00481193"; # Bumped to use holochain v0.0.131
```
Comment (zo-el) [inline overlays/holo-nixpkgs/configure-holochain/default.nix:10]:
```suggestion
    # Bumped to use holochain v0.0.131
    # Gets mem-proof for randomly generated keys in devnet
    rev = "aa64c168f8c13a6618bc6362e3a2688f00481193";
```

**Pull request merged**